src/App.tsx

```diff
@@ -86,7 +95,7 @@ export const App: React.FC = () => {
                   <PostsList
                     posts={posts}
                     selectedPostId={selectedPost?.id}
-                    onPostSelected={setSelectedPost}
+                    onPostSelected={selectPost}
```

Comment:
You have these in the store. Can't you just pull it from state in that component instead of passing it as a prop?